openstack/compute/v2/servers/requests.go

```diff
@@ -182,6 +182,7 @@ type CreateOpts struct {
 
 // ToServerCreateMap assembles a request body based on the contents of a CreateOpts.
 func (opts CreateOpts) ToServerCreateMap() (map[string]interface{}, error) {
+	var serverCreateMapError error
```

Comment:
Hmm. I see the issue now: I was thinking we shouldn't have to modify this at all, but we do because the request body hasn't been fully populated when the error occurs. Options include:
1. Removing the logic that checks for an image ID/name and let OpenStack handle it
2. Fake it by providing an `ImageID` if it and `ImageName` don't exist, call `servers.ToServerCreateMap`, then delete the `image_ref` map entry.
3. Move the `ImageID`/`ImageName` checking logic down to the bottom of `servers.ToServerCreateMap` such that the request body is completely populated and the `error` returned will be that one or `nil`.

What do you think of those options (or others)?